src/metabase/driver/sql_jdbc/common.clj

```diff
@@ -52,3 +56,19 @@
                                                                                                                :or   {seperator-style :url}}]
    (-> (dissoc connection-spec :additional-options)
        (assoc :subname (conn-str-with-additional-opts connection-string seperator-style additional-options)))))
+
+(defn parse-additional-options-value
+  "Attempts to parse the value for `opt-name` from an `additional-options` string, with `separator-style`, having
+  `opt-name-val-separator?` in between option names and values (optional; defaults to '='). Searches the additional
+  options string in a case insensitive manner."
+  [additional-options opt-name separator-style & [name-value-separator?]]
+  {:pre [(or (nil? additional-options) (string? additional-options))
+         (string? opt-name)
+         (contains? valid-separator-styles separator-style)
+         (or (nil? name-value-separator?) (and (string? name-value-separator?)
+                                               (= 1 (count name-value-separator?))))]}
+  (let [entry-sep (separator-style->entry-separator separator-style)
+        nv-sep    (or name-value-separator? default-name-value-separator)
+        re-pat    (str "(?i).*(?:^|" entry-sep ")(?:" opt-name nv-sep ")([^" nv-sep entry-sep "]+).*$")
+        [_ match] (re-matches (re-pattern re-pat) additional-options)]
+    match))
```

Comment:
This looks like a better version of `metabase.driver.h2/connection-string->file+options`... do you think we could rework that code to use this as well?

Reply:
Looked into it, and I think unfortunately the answer is "no".  That function actually handles the `file` value separately from `file:my-db;OPTION=100;OPTION_2=TRUE` (which has a different separator from its "value" as compared to the other params).  To shoehorn support for that into here would likely overcomplicate to the point of being no longer worth it.